src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+import { FilterTodos } from './types/FilterTodos';
+
+const filterTodos:FilterTodos = (todos, query, completedStatus) => {
+  let prepearedTodos = [...todos];
+
+  if (query.trim()) {
+    prepearedTodos = prepearedTodos.filter(todo => {
+      const normalizedTitle = todo.title.toLowerCase();
+      const normalizedQuery = query.toLowerCase().trim();
+
+      return normalizedTitle.includes(normalizedQuery);
+    });
+  }
+
+  prepearedTodos = prepearedTodos.filter(todo => {
+    switch (completedStatus) {
+      case TodoStatus.Active:
+        return !todo.completed;
+      case TodoStatus.Completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  });
+
+  return prepearedTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [completedStatus, setCompletedStatus] = useState<TodoStatus>(TodoStatus.All);
+  const [query, setQuery] = useState('');
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+
+  const getTodosFromServer = async () => {
+    const todosFromServer = await getTodos();
```

Comment:
if there will be any errors while creating api request - your app will be broken, let's fix it